Source/MediaInfo/Audio/File_Adm.cpp

```diff
@@ -2836,8 +2836,8 @@ static void CheckErrors_formatLabelDefinition(file_adm_private* File_Adm_Private
     const auto& Definition = Item.Attributes[label_Info.Definition_Pos];
     const auto Style = label_Info.Label_Style;
     const auto& Style_Info = Style_Infos[Style];
-    const auto List = (Style == (style)-1) ? nullptr : Style_Infos[Style].formatDefinition_List;
-    const unsigned long List_Size = (Style == (style)-1) ? 0 : Style_Infos[Style].formatDefinition_Size;
+    const auto List = (Style & (style)-1) ? nullptr : Style_Infos[Style].formatDefinition_List;

```

Comment:
It seems that it is wrongly seen as a power of two but the enum is:
```
enum style : int8u {
    Style_Format,
    Style_Type,
    Style_Max = (style)-1
};
```
a third style would be 3 so no more powers of 2.
this warning here is weird because Style_Max  is not a power of 2, I don't see how to remove the warning.
Hmmm... Actually a better code could be to remove Style_Max and (style)-1, and to put Style_None as the first element of the list.
checks would become (Style == Style_None) and the compiler will convert to "Style" , micro-optimization.